backend/rag_service.py

```
@@ -22,48 +25,67 @@ def __init__(self, policies_path: str = "backend/data/civic_policies.json"):
         self.policies = []
+        self._prepared_policies = []
         try:
             if os.path.exists(policies_path):
                 with open(policies_path, 'r') as f:
                     self.policies = json.load(f)
-                logger.info(f"Loaded {len(self.policies)} civic policies for RAG.")
+                self._prepare_policies()
+                logger.info(f"Loaded and prepared {len(self.policies)} civic policies for RAG.")
             else:
                 logger.warning(f"Civic policies file not found at {policies_path}")
         except Exception as e:
```

Ruff (0.15.10)

[warning] Do not catch blind exception: `Exception`

(BLE001)